tests/framework/extensions/machinepools/machinepools.go

```diff
@@ -80,3 +129,68 @@ func RKEMachinePoolSetup(nodeRoles []NodeRoles, machineConfig *v1.SteveAPIObject
 
 	return machinePools
 }
+
+// ScaleWorkerMachinePool is a helper method that will add a worker node pool to the existing RKE2/K3S cluster. Once done, it will scale
+// the worker machine pool to add a worker node, scale it back down to remove the worker node, and then delete the worker machine pool.
+func ScaleNewWorkerMachinePool(client *rancher.Client, cluster *v1.SteveAPIObject, updatedCluster *apisV1.Cluster, machineConfig *v1.SteveAPIObject) error {
```

Comment:
This function is very specific to the validation we are testing. Why don't you make this more generic so we can test a variety of different scaling scenarios.